bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -279,6 +262,59 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return r.joinWorker(ctx, scope)
 }
 
+func (r *KubeadmConfigReconciler) refreshBootstrapToken(ctx context.Context, config *bootstrapv1.KubeadmConfig, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	token := config.Spec.JoinConfiguration.Discovery.BootstrapToken.Token
+
+	remoteClient, err := r.remoteClientGetter(ctx, r.Client, util.ObjectKey(cluster))
+	if err != nil {
+		log.Error(err, "Error creating remote cluster client")
+		return ctrl.Result{}, err
+	}
+
+	log.Info("Refreshing token until the infrastructure has a chance to consume it")
+	err = refreshToken(ctx, remoteClient, token)
+	if err != nil {
+		// It would be nice to re-create the bootstrap token if the error was "not found", but we have no way to update the Machine's bootstrap data
+		return ctrl.Result{}, errors.Wrapf(err, "failed to refresh bootstrap token")
+	}
+	// NB: this may not be sufficient to keep the token live if we don't see it before it expires, but when we generate a config we will set the status to "ready" which should generate an update event
```

Comment:
Argh requeue afters 😄, I'm not sure if I follow this comment either, if we renewed the token and we requeue at half the TTL, we should be fine?

Reply:
same comment as above (was taken from https://github.com/kubernetes-sigs/cluster-api/pull/3940/files#diff-f754cabdd737c0fac3039e7eb563b50ea52fc4ff88a9d517f4067e25977acdadL248)